try.go

```diff
@@ -70,62 +69,150 @@
+//
+//
+// Handle allows easy assignment of that error to a return error value.
+//
+//  func f() (err error) {
+//  	defer try.Handle(&err)
+//
+//
+// HandleF is like Handle, but it calls a function after any such assignment.
+//
+//  func f() (err error) {
+//		defer try.HandleF(&err, func() {
+//			if err == io.EOF {
+//				err = io.ErrUnexpectedEOF
+//			}
+//		})
+//
+// F wraps an error with file and line formation and calls a function on error.
+// It plays nicely with testing.TB and log.Fatal.
+//
+//  func TestFoo(t *testing.T) {
+//  	defer try.F(t.Fatal)
+//
+//  func main() {
+//  	defer try.F(log.Fatal)
+//
+//
+// Recover is like F, but it supports more complicated error handling
+// by passing the error and runtime frame directly to a function.
+//
+//  func f() {
+//  	defer try.Recover(func(err error, frame runtime.Frame) {
+//  		// do something useful with err and frame
+// 		})
+//
+//
 package try
 
+import (
+	"fmt"
+	"runtime"
+)
+
 // wrapError wraps an error to ensure that we only recover from errors
 // panicked by this package.
-type wrapError struct{ error }
+type wrapError struct {
+	error
+	frame runtime.Frame
+}
 
 // Unwrap primarily exists for testing purposes.
-func (e wrapError) Unwrap() error { return e.error }
+func (e wrapError) Unwrap() error {
+	return e.error
+}
 
-// Catch catches a previously panicked error and stores it into err.
-// If it successfully catches an error, it calls any provided handlers.
-func Catch(err *error, handlers ...func()) {
-	switch ex := recover().(type) {
+func r(recovered any, fn func(err error, frame runtime.Frame)) {
+	switch ex := recovered.(type) {
 	case nil:
-		return
 	case wrapError:
-		*err = ex.error
-		for _, handler := range handlers {
-			handler()
-		}
+		fn(ex.error, ex.frame)
 	default:
 		panic(ex)
 	}
 }
 
-// E panics if err is non-nil.
-func E(err error) {
+// Recover recovers an error previously panicked with an E function.
+// If it recovers an error, it calls fn with the error and the runtime frame in which it occurred.
+func Recover(fn func(err error, frame runtime.Frame)) {
+	r(recover(), fn)
+}
+
+// Handle recovers an error previously panicked with an E function and stores it into errptr.
+func Handle(errptr *error) {
+	r(recover(), func(err error, _ runtime.Frame) {
+		*errptr = err
+	})
+}
+
+// HandleF recovers an error previously panicked with an E function and stores it into errptr.
+// If it recovers an error, it calls fn.
+func HandleF(errptr *error, fn func()) {
+	r(recover(), func(err error, _ runtime.Frame) {
+		*errptr = err
+		if err != nil {
+			fn()
+		}
+	})
+}
+
+// F recovers an error previously panicked with an E function, wraps it, and passes it to fn.
+// The wrapping includes the file and line of the runtime frame in which it occurred.
+// F pairs well with testing.TB.Fatal and log.Fatal.
+func F(fn func(...any)) {
+	r(recover(), func(err error, frame runtime.Frame) {
+		fn(fmt.Errorf("%s:%d: %w", frame.File, frame.Line, err))
```

Comment:
This is the only usage of `fmt`, which will cause a transitive dependency on `reflect`, `unicode`, `sort`, `strconv`, and many other packages.

How about we just change `r` to take in a `func(*wrapError)` and directly call `fn` with the `*wrapError`. Of course, we'll need to add a `wrapError.Error` method that does the `file:line` prefix.

Reply:
Done; now the dep is only `strconv`.